### PR TITLE
Change `#[insertable_into]` to `#[derive(Insertable)]`

### PR DESCRIPTION
**Pull request by sgrif**
This makes our API compatible with something that can be represented in
Macros 1.1

Review? @diesel-rs/core

### CODE REVIEW

File: diesel/src/persistable.rs [hunk 1]
```diff
@@ -6,11 +6,10 @@ use result::QueryResult;
 use query_builder::{QueryBuilder, BuildQueryResult};
 use query_source::{Table, Column};
 
-/// Represents that a structure can be used to to insert a new row into the database.
-/// Implementations can be automatically generated by
-/// [`#[insertable_into]`](https://github.com/sgrif/diesel/tree/master/diesel_codegen#insertable_intotable_name).
-/// This is automatically implemented for `&[T]` and `&Vec<T>` for inserting more than
-/// one record.
+/// Represents that a structure can be used to to insert a new row into the
+/// database.  Implementations can be automatically generated by
```
Comment (killercup) [inline diesel/src/persistable.rs]: double space

File: diesel/src/persistable.rs [hunk 2]
```diff
@@ -6,11 +6,10 @@ use result::QueryResult;
...
+/// `#[derive(Insertable)]` This is automatically implemented for `&[T]` and
```
Comment (killercup) [inline diesel/src/persistable.rs]: Add period before 'This'

File: diesel_codegen/README.md [hunk 3]
```diff
@@ -93,17 +93,20 @@ Adds an implementation of the [`Queryable`][queryable] trait to the annotated
...
 additional configurations.
 
+Structs which derive `Insertable` must have a table name annotation like so:
+`#[table_name = "something"]`
+
 ### `#[derive(AsChangeset)]`
 
 Adds an implementation of the [`AsChangeset`][as_changeset] trait to the
-annotated item, targeting the given table. At this time, it only supports
+annotated item, targeting the given table. at this time, it only supports
```
Comment (killercup) [inline diesel_codegen/README.md]: Did you mean to change this sentence in another way?

File: diesel_codegen/src/lib.rs [hunk 4]
```diff
@@ -23,12 +23,12 @@ pub fn register(reg: &mut rustc_plugin::Registry) {
...
-        intern("derive_Queryable"),
-        MultiDecorator(Box::new(queryable::expand_derive_queryable))
+        intern("derive_Insertable"),
+        MultiDecorator(Box::new(insertable::expand_derive_insertable))
     );
     reg.register_syntax_extension(
-        intern("insertable_into"),
-        MultiDecorator(Box::new(insertable::expand_insert))
+        intern("derive_Queryable"),
+        MultiDecorator(Box::new(queryable::expand_derive_queryable))
```
Comment (killercup) [inline diesel_codegen/src/lib.rs]: Is there a reason you switched the order of the register calls (insert before query)? Same in `diesel_codegen_syntex/src/lib.rs`. It obviously works but I couldn't find a reason if the order makes a difference.

File: diesel_codegen_syntex/src/insertable.rs [hunk 5]
```diff
@@ -1,84 +1,42 @@
...
-        }
-        _ => usage_error(cx, meta_item),
-    }
-}
-
-fn table_name(cx: &mut ExtCtxt, meta_item: &MetaItem) -> InternedString {
-    match meta_item.node {
-        MetaItemKind::Word(ref word) => word.clone(),
-        _ => usage_error(cx, meta_item),
+    if let Some(model) = Model::from_annotable(cx, span, annotatable) {
```
Comment (killercup) [inline diesel_codegen_syntex/src/insertable.rs]: Oh, nice. You are basically using the logic you already have in `Model` to do the heavy lifting, right? Actually—is this the exact same code?

File: diesel_codegen/src/lib.rs (hunk 4 above)
Comment (sgrif) [inline diesel_codegen/src/lib.rs]: Keeping them alphabetical

**Pull request merged**